br/pkg/lightning/restore/restore.go

```diff
@@ -1236,7 +1255,54 @@ func (rc *Controller) buildRunPeriodicActionAndCancelFunc(ctx context.Context, s
 
 var checksumManagerKey struct{}
 
+const (
+	pauseGCTTLForDupeRes       = time.Hour
+	pauseGCIntervalForDupeRes  = time.Minute
+	pauseGCServiceIDForDupeRes = "lightning-duplicate-resolution"
+)
+
+func (rc *Controller) pauseGCOnce(ctx context.Context, pdCli pd.Client) error {
+	// Get current GC safe point.
+	safePoint, err := pdCli.UpdateGCSafePoint(ctx, 0)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// Pause GC for 1 hour.
+	_, err = pdCli.UpdateServiceGCSafePoint(ctx,
+		pauseGCServiceIDForDupeRes, int64(pauseGCTTLForDupeRes/time.Second), safePoint)
+	return err
+}
+
+func (rc *Controller) keepPauseGC(ctx context.Context, pdCli pd.Client) {
+	ticker := time.NewTicker(pauseGCIntervalForDupeRes)
+	defer ticker.Stop()
```

Comment:
Do we need to remove this service GC safepoint after this function finished?

Reply:
No. If there are multiple lightnings, `duplicate-resolution` expects GC is paused all the time.

Reply:
We should make each lightning instance a separate service, so their gc ttl are separated too. Then we can safely cancel the ttl after each lightning finished.

Reply:
Or should we use a unique service ID for each Lightning instance.